pkg/operator/podsecurityreadinesscontroller/podsecurityreadinesscontroller.go

```diff
@@ -0,0 +1,176 @@
+package podsecurityreadinesscontroller
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	psapi "k8s.io/pod-security-admission/api"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+const (
+	checkInterval = 10 * time.Minute // Adjust the interval as needed.
+)
+
+var (
+	// run-level zero namespaces, shouldn't avoid openshift namespaces
+	ignoredNamespaces = map[string]struct{}{
+		"default":     empty,
+		"kube-system": empty,
+		"kube-public": empty,
+	}
+	empty                  = struct{}{}
+	podSecurityAlertLabels = []string{
+		psapi.AuditLevelLabel,
+		psapi.WarnLevelLabel,
+	}
+)
+
+// PodSecurityReadinessController checks if namespaces are ready for Pod Security Admission enforcement.
+type PodSecurityReadinessController struct {
+	kubeClient     kubernetes.Interface
+	operatorClient v1helpers.OperatorClient
+
+	warningsHandler *warningsHandler
+}
+
+func NewPodSecurityReadinessController(
+	kubeConfig *rest.Config,
+	operatorClient v1helpers.OperatorClient,
+	recorder events.Recorder,
+) (factory.Controller, error) {
+	warningsHandler := &warningsHandler{}
+	kubeClientCopy := rest.CopyConfig(kubeConfig)
+	kubeClientCopy.WarningHandler = warningsHandler
+	kubeClient, err := kubernetes.NewForConfig(kubeClientCopy)
+	if err != nil {
+		return nil, err
+	}
+	c := &PodSecurityReadinessController{
+		operatorClient:  operatorClient,
+		kubeClient:      kubeClient,
+		warningsHandler: warningsHandler,
+	}
+
+	return factory.New().
+		WithSync(c.sync).
+		ResyncEvery(checkInterval).
+		ToController("PodSecurityReadinessController", recorder), nil
+}
+
+func (c *PodSecurityReadinessController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	nsList, err := c.kubeClient.CoreV1().Namespaces().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	violatingNamespaces := []string{}
+	for _, ns := range nsList.Items {
+		isViolating, err := c.isNamespaceViolating(ctx, ns)
+		if err != nil {
+			return err
+		}
+		if isViolating {
+			violatingNamespaces = append(violatingNamespaces, ns.Name)
+		}
+	}
+
+	cond := operatorv1.OperatorCondition{
+		Type:               string(configv1.EvaluationConditionsDetected),
```

Comment:
do you perhaps want to customize the type a bit to be more explicit in the message? Like `PodSecurityReadinessControllerEvaluationConditionsDetected` and have the reason be something more specific like: `NamespacesAreReadyForPodSecurityEnforcement`?